lib/Driver/Driver.cpp

```diff
@@ -217,14 +217,17 @@ phases::ID Driver::getFinalPhase(const DerivedArgList &DAL,
   phases::ID FinalPhase;
 
   // -{E,EP,P,M,MM} only run the preprocessor.
-  if (CCCIsCPP() || (PhaseArg = DAL.getLastArg(options::OPT_E)) ||
+  // Do not use the clang preprocessor in fortran mode with -E
+  if (CCCIsCPP() || (!IsFortranMode() &&
+      (PhaseArg = DAL.getLastArg(options::OPT_E))) ||
       (PhaseArg = DAL.getLastArg(options::OPT__SLASH_EP)) ||
```

Comment:
Why aren't we handling SLASH_EP, M, MM, or SLASH_P? Shouldn't those work for flang like clang?